### ISSUE TITLE
Saving profiles Issue

### ISSUE THREAD

**Issue by Neob0b** [magicalcookie/jc3-handling-editor#10]
Hi!

The spawn and edit menus work pretty well, but the saving part makes my game crash instantly.
Same issue with "Save" or "Save as".
No problem to enter a name for the profile, but the crash happens when hitting the button "save".

Edit : It seems that it creates a "profiles" folder in my "Mods" folder (JC3 installation path).
Then it creates subfolders named as the game object edited.
In the subfolder appears only a ".json" fil that is empty.

Any Solution?

Thx

**Comment by Neob0b** [comment]
Hello,

So after some tests I have some news. The problem happens with boats, planes and hilcopters. I didn't notice it before, but Iit seems that cars and mtorbikes work pretty well.

So I tried to investigate a bit, and create a ".json" profile that can be loaded in the game for the rebel' s fighterjet:

`{
  "topSpeed": 500,

  "Air Engine": {
    "max_thrust": 80000.000,
    "max_thrust_acceleration": 80000.000,
    "min_thrust": 10000.000,
    "run_thrust": 40000.000,
    "taxiing_input_thrust": 0.9,
    "taxiing_max_thrust": 20000.000,
    "taxiing_top_speed": 20.000
  },

  "Air Steering": {

    "referenceMaxSpeedKPH": 360,
    "referenceMinSpeedKPH": 36,
    "min_thrust": 0.000,
    "run_thrust": 0.600,
    "pitch_return": 0.000,
    "roll_return": 0.000,

    "rollAxisTiming": {
      "timeToMaxInputAtMinSpeed_s": 1.000,
      "timeToMaxInputAtMaxSpeed_s": 1.200,
      "centeringInputTimeFactor": 1.200,
      "counterInputTimeFactor": 5.000
    },
    "pitchAxisTiming": {
      "timeToMaxInputAtMinSpeed_s": 1.000,
      "timeToMaxInputAtMaxSpeed_s": 1.000,
      "centeringInputTimeFactor": 1.200,
      "counterInputTimeFactor": 5.000
    },

    "yawAxisTiming": {
      "timeToMaxInputAtMinSpeed_s": 1.000,
      "timeToMaxInputAtMaxSpeed_s": 1.000,
      "centeringInputTimeFactor": 1.000,
      "counterInputTimeFactor": 3.000
    },
  }`

I just copied the code stucture that appears in other .json profiles, and created lines with the parameters that are editable in the game.

It didn't work.

Then I read the "jc3_handling_editor.cpp", but didn't notice that only cars and motorbikes can work with the code.

I wanted to do tests by editing the code, but I don't know how to generate a dll file from the full folder of pieces of code.

Can someone tell me if I'm the only one to encounter this problem?
Or is it normal because the editor is no made to edit unterrestrian vehicles?

Please help me to fix it. I admit to be a noob  in code languages.
I won't be able to go on just with motivation at this point :/

Thx

**Comment by xforce** [comment]
Yes profile support is not really implemented yet, I do plan on finishing this,  wanted to get it done months ago but then I had a lot of stuff to do for work and the multiplayer mod so this little side project was at the bottom of my priority list.

I will try to get something done once we finished 0.9.9 for our multiplayer mod.

**Comment by Neob0b** [comment]
Hi xforce.

Thank you for your answer.
I better understand now.
I would like to suggest you my help, but I think it will more bother you than help, considering my level.
